### ISSUE TITLE
New Guide: Managing Drupal Core and Contrib with Composer (and without an external repo)

### ISSUE THREAD

**Issue by stevector** [pantheon-systems/documentation#3251]
Re: [Composer](https://pantheon.io/docs/composer/)

## Issue Description

Using Terminus Build Tools with an external repo (like GitHub) and a 3rd party CI service (like CircleCI) is more overhead than many Pantheon customers want while still using Composer with Drupal 8.

## Suggested Resolution

Write a guide documenting how to set up and use a copy[ example-drops-8-composer](https://github.com/pantheon-systems/example-drops-8-composer) without putting it on GitHub.

This issue came up in Power Users and I wrote an outline of what the guide could be. https://groups.google.com/a/pantheon.io/d/msg/power-users/Yf7su41Shrc/LGnkY2GHBQAJ

**Comment by ummdorian** [comment]
I have heard developers say they don't use Pantheon because they thought composer managed core was not possible without third party software.  I personally wasted a lot of time trying to figure it out on my own.  I think this is critical documentation that needs to replace the current composer managed core documentation.

**Comment by ummdorian** [comment]
Also, if it's possible to transition a currently Pantheon hosted D8 site to manage core with composer without migrating to a new install a walk-through or automation tools for that would be helpful.

**Comment by ummdorian** [comment]
Missing from the above linked walkthrough is changing the config directory per:
https://pantheon.io/docs/drupal-8-configuration-management/
```
$config_directories = array(
  CONFIG_SYNC_DIRECTORY => dirname(DRUPAL_ROOT) . '/config',
);
```

**Comment by klrodgers** [comment]
@mcdwayne suggested that I try this doc during yesterday's Office Hours.  It is really well done, and I think it will be valuable.  I would want CI eventually, but for now, I just need to get up quickly with the Apigee plugin for d8.  It is documented assuming composer and is in an alpha state, so this guide was just what I needed.

I have 2 feedbacks:
1. I went thru twice on my usual windows box, but the `terminus drush ... -- site-install -y` step failed with `[error]` but no following text.  I'm new to pantheon, so I wasn't sure how to debug that, and didn't immediately find a relevant search result.  Today I found an Ubuntu to try on and everything went thru flawlessly.   Probably something was wrong in my setup on windows, but for a while I was stuck.
2. When I get to the end of the guide, I still have a development branch and a multidev instance running.  I suppose I should merge locally and push the master branch, then kill the multidev with some terminus command?    It would be nice to make that explicit--either with steps, or just some high-level text description.

Thanks again for this guide!
